sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/_ownership_manager.py

```diff
@@ -0,0 +1,135 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+
+import time
+import random
+import math
+from collections import Counter
+from azure.eventhub.aio import EventHubClient
+
+
+class OwnershipManager(object):
+    """Increases or decreases the number of partitions owned by an EventProcessor
+    so the number of owned partitions are balanced among multiple EventProcessors
+
+    An EventProcessor calls claim_ownership() of this class every x seconds,
+    where x is set by keyword argument "polling_interval" in EventProcessor,
+    to claim the ownership of partitions, create tasks for the claimed ownership, and cancel tasks that no longer belong
+    to the claimed ownership.
+
+    """
+    def __init__(self, event_processor, eventhub_client: EventHubClient, ownership_timeout: int):
+        self.all_parition_ids = []
```

Comment:
rename to `self._cached_partition_ids`?

Reply:
"cached" is more like a technical concept. It could mean all or a part of something. In this context, I prefer "all" or any other words that mean this is all the partition ids. It's true that they happen to be cached.
Someday the number of partitions can increase after an eventhub is created. Then the partition ids should be cached.

Reply:
If they weren't cached, we wouldn't have the attribute - we would just have a property or method to retrieve the list. So I'm not concerned about caching being an internal name - it is more indicative of what it actually is, and thus more likely to be fixed if there was a problem.

Reply:
Change name to self._cached_partition_ids since for now there is no requirement to have changeable partition ids